pallene/ir.lua

```diff
@@ -114,14 +130,16 @@ end
 --
 
 declare_type("Value", {
-    Nil        = {},
-    Bool       = {"value"},
-    Integer    = {"value"},
-    Float      = {"value"},
-    String     = {"value"},
-    LocalVar   = {"id"},
-    Upvalue    = {"id"},
-    Function   = {"id"},
+    Nil                = {},
+    Bool               = {"value"},
+    Integer            = {"value"},
+    Float              = {"value"},
+    String             = {"value"},
+    LocalVar           = {"id"},
+    Upvalue            = {"id"},
+    Function           = {"id"},
+    ImportedFunction   = {"id"},
```

Comment:
Silly question: why do we need separate cases for the Function and ImportedFunction?

Reply:
I tried using `Function` at first and adding the imported functions to the `module.functions` table, but there are several places in the code that loop through `module.functions` and expect there to be a body field. Since we are not running the compiler past the `Checker` phase ( for obvious reasons),  the imported functions won't have a body and, thus, the checks will fail. In hindsight, I think it would be possible to stick with `ir.Value.Function` and just insert the imported functions into an `imported_functions` array. I'll see if it's actually possible.

Reply:
If the imported functions work differently, I think it is ok to have a separate case for them.